test/Application.UTest/Items/BuyItemCommandTest.cs

```diff
@@ -14,13 +14,14 @@ public class BuyItemCommandTest : TestBase
     [Test]
     public async Task Basic()
     {
-        Item item = new() { Id = "0", Price = 100, Enabled = true };
+        Item item = new() { Id = "0", Price = 100, Enabled = true, Rank = 0 };
         User user = new()
         {
             Gold = 100,
-            Items = { new UserItem { Rank = 1, Item = item } },
+            Items = { },
         };
         ArrangeDb.Users.Add(user);
+        ArrangeDb.Items.Add(item);
```

Comment:
You should only move the rank here and discard the rest. The test was checking that you are allowed to have a rank 0 and 1 of the same item.

Reply:
Moving the rank here does not work and causes the test to fail as with a single item , you cannot have two different ranks. There is no such thing as the same item having the different ranks in the new system.
This test also check that everything works alright (broken status , gold reduction , item bought is the one received.)
If you want me to do it another way , please advise a way that would not cause the test to fail.

Reply:
The heirloomed version as the same baseid but a different id. The non heirloomed version end in h0 et the heirloomed one in h1 h2 h3.
In the old version there was a single Item , but different UserItem as the rank was stored in UserItem.
That's why you cannot have the same item (crpg_claymore_h0) in different ranks. It's h0 so it is rank0 . The rank 1 is another item (crpg_claymore_h1) .
But this test was written for the old sytem where it was item (crpg_claymore) , and Useritem (id = crpg_claymore, Rank = 1)

Reply:
I could rewrite the test with two items , but they would have different id 
 Item item0 = new() { Id = a_h0", Price = 100, Enabled = true, Rank = 0 };
 Item item1 = new() { Id = "a_h1", Price = 100, Enabled = true, Rank = 1 };
        User user = new()
        {
            Gold = 100,
            Items = { new UserItem { Item = item1 }},
        }
But the test would not really make sense

Reply:
No it makes sense to me. The test was checking if I own a rank 1 item, can I buy a rank 0 of the same type.

Reply:
> No it makes sense to me. The test was checking if I own a rank 1 item, can I buy a rank 0 of the same type.

it's done . But i did not remove lines , i had to add ones .
the lines 
```
ArrangeDb.Items.Add(item0);
ArrangeDb.Items.Add(item1);
```
are needed otherwise you get this error
![image](https://github.com/verdie-g/crpg/assets/47090987/36af887f-7051-4339-9424-90f0eeac491c)